lib/dynflow/action/polling.rb

```diff
@@ -13,6 +14,9 @@ def run(event = nil)
         end
       when Poll
         poll_external_task_with_rescue
+      when Timeout
+        process_timeout
+        poll_external_task_with_rescue
```

Comment:
Only if process timeout throws an exception (in the default case that is true, but in the 'advanced' cases where process_timeout is overridden, maybe not).